eodhp_utils/pulsar/client.py

```diff
@@ -0,0 +1,52 @@
+import json
+import os
+
+from pulsar import Client as PulsarClient
+
+
+class Client:
+    def __init__(self, url=None):
+        if url is None:
+            url = os.environ.get("PULSAR_URL")
+        self.client = PulsarClient(url)
+
+
+class Producer(Client):
+    def __init__(self, topic, producer):
```

Comment:
annotate both of these inputs as str please. It's slightly confusing that a user passes in "producer" as a str but then self.producer is an object. I would match up the input parameter to your init to the create_producer parameter and use producer_name.

Reply:
You also need to pass through pulsar_url to the subclass init function and not rely on env vars.